runtime-light/stdlib/component/inter-component-session/buffer-provider.h

```diff
@@ -0,0 +1,109 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2025 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include "runtime-common/core/allocator/script-malloc-interface.h"
+
+namespace kphp::component::inter_component_session {
+
+class BufferProvider final {
```

Comment:
Can you motivate your decision to add such a functionality? I think we way try to replace it with something more standard